cypress/integration/Header.spec.js

```diff
@@ -0,0 +1,110 @@
+const viewports = [
+  // Desktop
+  {
+    size: [1440, 900],
+  },
+  // Large mobile phone
+  {
+    preset: 'samsung-s10',
+    orientation: 'portrait',
+  },
+  {
+    preset: 'samsung-s10',
+    orientation: 'landscape',
+  },
+  // Small mobile phone
+  {
+    preset: 'iphone-3',
+    orientation: 'portrait',
+  },
+  {
+    preset: 'iphone-3',
+    orientation: 'landscape',
+  },
+  // Tablet
+  {
+    preset: 'ipad-2',
+    orientation: 'portrait',
+  },
+  {
+    preset: 'ipad-2',
+    orientation: 'landscape',
+  },
+];
+
+const setViewPort = (viewport) => {
+  if (viewport.size) {
+    cy.viewport(viewport.size[0], viewport.size[1]);
+  } else {
+    cy.viewport(viewport.preset, viewport.orientation);
+  }
+};
+
+const getViewPortName = (viewport) => {
+  if (viewport.size) {
+    return viewport.size;
+  }
+  return `${viewport.preset}, ${viewport.orientation}`;
+};
+
+describe('Header components', () => {
+  beforeEach(() => {
+    cy.visit('');
+  });
+
+  // For each viewport
+  viewports.forEach((viewport) => {
+    describe(`${getViewPortName(viewport)} screen`, () => {
+      it(`should display menu`, () => {
+        setViewPort(viewport);
+        cy.get('.wkp-menu-header').should('be.visible');
+        cy.get('.wkp-topics-menu-body').should('not.be.visible');
+        cy.get('.wkp-menu-header').click();
+
+        // Menu should open
+        cy.get('.wkp-topics-menu-body').should('be.visible');
+      });
+
+      it(`should display WKP login`, () => {
+        setViewPort(viewport);
+        cy.get('.wkp-login').should('be.visible');
+        // Do not test click because of CORS problem
+        // when redirecting to another page
+      });
+
+      it(`should display language select`, () => {
+        setViewPort(viewport);
+        cy.get('.wkp-single-value-wrapper').should('be.visible');
+
+        // Click the react-select element
+        cy.get('.wkp-header .css-m869aq-ValueContainer').click();
```

Comment:
Isn't this class name gonna change on build? Maybe with an id we could select the correct item, else it is good enough.

Reply:
@edellucien has found a workaround for the IABP project where a ValueContainer and Menu is defined for the react-select component, additionally to the SingleContainer. So, we have  beautiful css names. Unfortunately, this does not work in trafimage-maps. And we don't know why. We get a detached-from-DOM error. And even, if we do not change the classes in the test, it does not work anymore.

I have checked if the strange react-select css classes change after a built or when the react-select version changes - no. So, I prefer to leave this test the way it is. @edellucien agrees.